kafka_consumer/datadog_checks/kafka_consumer/client/generic_kafka_client.py

```diff
@@ -0,0 +1,80 @@
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.get_consumer_offsets()
+        # return self.confluent_kafka_client.get_consumer_offsets()
+
+        return self.python_kafka_client.get_consumer_offsets()
+
+    def get_highwater_offsets(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.get_highwater_offsets()
+        # return self.confluent_kafka_client.get_highwater_offsets()
+        return self.python_kafka_client.get_highwater_offsets()
+
+    def get_highwater_offsets_dict(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.get_highwater_offsets_dict()
+        # return self.confluent_kafka_client.get_highwater_offsets_dict()
+        return self.python_kafka_client.get_highwater_offsets_dict()
+
+    def reset_offsets(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.reset_offsets()
+        # return self.confluent_kafka_client.reset_offsets()
+        return self.python_kafka_client.reset_offsets()
+
+    def get_partitions_for_topic(self, topic):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.get_partitions_for_topic(topic)
+        # return self.confluent_kafka_client.get_partitions_for_topic(topic)
+        return self.python_kafka_client.get_partitions_for_topic(topic)
+
+    def request_metadata_update(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.request_metadata_update()
+        # return self.confluent_kafka_client.request_metadata_update()
+        return self.python_kafka_client.request_metadata_update()
+
+    def collect_broker_version(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.collect_broker_version()
+        # return self.confluent_kafka_client.collect_broker_version()
+        return self.python_kafka_client.collect_broker_version()
+
+    def get_consumer_offsets_dict(self):
+        # TODO when this method is implemented in ConfluentKafkaClient, replace this with:
+        # if self.use_legacy_client:
+        #     return self.python_kafka_client.get_consumer_offsets_dict()
+        # return self.confluent_kafka_client.get_consumer_offsets_dict()
+        return self.python_kafka_client.get_consumer_offsets_dict()
+
+    def create_kafka_admin_client(self):
```

Comment:
FYI I think we may need to remove `create_kafka_admin_client()` in the `kafka_python_client.py` code eventually since that was vestigial from the legacy implementation but couldn't outright remove due to it being used in the tests. I'm ok with keeping this for now, but I don't think we need to implement a similar function in `confluent_kafka_client`.

Reply:
Yeah I agree. I did not do it in this PR to focus on the global test structure. We can remove it on follow-up PRs :)